Media/MediaPlayer.cpp

```diff
@@ -1303,14 +1306,43 @@ int AudioBufferDataSource::read_packet(void *opaque, uint8_t *buf,
 int AudioBufferDataSource::ReadPacket(uint8_t *buf, int buf_size) {
     int size = buf_end - buf_pos;
     if (size <= 0) {
-        return 0;
+        return AVERROR_EOF;
     }
     size = std::min(buf_size, size);
     memcpy(buf, buf_pos, size);
     buf_pos += size;
     return size;
 }
 
+int64_t AudioBufferDataSource::seek(void *opaque, int64_t offset, int whence) {
+    AudioBufferDataSource *pThis = (AudioBufferDataSource *)opaque;
+    return pThis->Seek(opaque, offset, whence);
+}
+
+int64_t AudioBufferDataSource::Seek(void *opaque, int64_t offset, int whence) {
+    if ((whence & AVSEEK_SIZE) == AVSEEK_SIZE) {
+        return buffer->GetSize();
+    }
+    int force = whence & AVSEEK_FORCE;
+    whence &= ~AVSEEK_FORCE;
+    whence &= ~AVSEEK_SIZE;
+    uint8_t *buf_start = (uint8_t*)buffer->GetData();
+    if (whence == SEEK_SET) {
+        buf_pos = std::min(buf_start + offset, buf_end);
+        buf_pos = std::max(buf_pos, buf_start);
+        return buf_pos - buf_end;
```

Comment:
This doesn't seem to be a correct return value. Should be buf_pos - (uint8_t*)buffer->GetData()? Likely makes sense to store buf_start as a field.